src/libcollections/string.rs

```diff
@@ -701,6 +701,13 @@ impl String {
     /// Violating these may cause problems like corrupting the allocator's
     /// internal datastructures.
     ///
+    /// The ownership of `ptr` is effectively transferred to the
+    /// `String` and the memory referred to by the pointer will be
+    /// deallocated when the `String` goes out of scope. Ensure that
+    /// nothing else uses the pointer after calling this
+    /// function. Calling this function multiple times with the same
+    /// pointer is an error.
```

Comment:
This is difficult... You are absolutely technically correct. My original intent was to provide a warning for the developers who aren't at your proficiency level because I have seen them use this method incorrectly.

The other case this is incorrect is if you repeatedly cycle between `into_raw_parts` and `from_raw_parts` ;the pointer may indeed be the same.

Do you have any suggestions for a rewording that wouldn't be *incorrect* but also doesn't get too far off into the weeds?